src/layout.rs

```diff
@@ -527,25 +520,25 @@ fn try_split(area: Rect, layout: &Layout) -> Result<Rc<[Rect]>, AddConstraintErr
     }
     // ensure the first element touches the left/top edge of the area
     if let Some(first) = elements.first() {
-        solver.add_constraint(first.start | EQ(REQUIRED) | area.start)?;
+        solver.add_constraint(first.start | EQ(REQUIRED) | start)?;
     }
     // ensure the last element touches the right/bottom edge of the area
     if layout.expand_to_fill {
         if let Some(last) = elements.last() {
-            solver.add_constraint(last.end | EQ(REQUIRED) | area.end)?;
+            solver.add_constraint(last.end | EQ(REQUIRED) | end)?;
         }
     }
     // apply the constraints
     for (&constraint, &element) in layout.constraints.iter().zip(elements.iter()) {
         match constraint {
             Constraint::Percentage(p) => {
                 let percent = f64::from(p) / 100.00;
-                solver.add_constraint(element.size() | EQ(STRONG) | (area.size() * percent))?;
+                solver.add_constraint(element.size() | EQ(STRONG) | ((end - start) * percent))?;
```

Comment:
I assuming there are no edge cases where we need `checked_sub` here?

Reply:
Resolved in https://github.com/ratatui-org/ratatui/pull/411/commits/7693ec29e16c606c07c8d6dd0720e07025b696ec.